File: api/src/main/java/com/ankol/api/entity/ClanMember.java
package com.ankol.api.entity;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@Data
public class ClanMember {

    private String tag;
    private String name;
    private String role;
    private Integer expLevel;
    private LeagueDTO league;
    private Integer trophies;
    private Integer versusTrophies;
    private Integer clanRank;
    private Integer previousClanRank;
    private Integer donations;
    private Integer donationsReceived;
    private PlayerHouseDTO playerHouse;

    @NoArgsConstructor
    @Data
    public static class LeagueDTO {
        private Integer id;
        private String name;
        private IconUrlsDTO iconUrls;

        @NoArgsConstructor
        @Data
        public static class IconUrlsDTO {
            private String small;
            private String tiny;
            private String medium;
        }
    }

    @NoArgsConstructor
    @Data
    public static class PlayerHouseDTO {
        private List<ElementsDTO> elements;

        @NoArgsConstructor
        @Data
        public static class ElementsDTO {
            private String type;
            private Integer id;
        }
    }
}

File: api/src/main/java/org/example/api/pojo/ClanMember.kt
package org.example.api.pojo

/**
 * 成员
 */
data class ClanMember(
    /**
     * 标签
     */
    val tag: String,
    /**
     * 成员名称
     */
    val name: String,
    /**
     * 成员职位
     */
    val role: String,
    /**
     * 经验等级
     */
    val expLevel: Short,

    val league: Any,
    /**
     * 奖杯
     */
    val trophies: Short,
    /**
     * 对抗赛奖杯
     */
    val versusTrophies: Short,
    val clanRank: Short,
    val previousClanRank: Short,
    /**
     * 捐兵数
     */
    val donations: Int,
    /**
     * 收兵数
     */
    val donationsReceived: Short,
)
